src/views/Home/components/DaysLeft/index.jsx

```diff
@@ -24,65 +33,145 @@ const DaysLeft = () => {
       ]);
       const first = format(parseISO(currentSession.SessionBeginDate), 'MM/dd/yyyy');
       const last = format(parseISO(currentSession.SessionEndDate), 'MM/dd/yyyy');
-      const currentSessionDescription = currentSession.SessionDescription.replace(
-        /(Academic Year)|(Grad)/gm,
-        '',
-      );
+      const fallBegin = new Date((await session.get(`${currentYear - 1}09`)).SessionBeginDate);
+      const fallEnd = new Date((await session.get(`${currentYear - 1}09`)).SessionEndDate);
+      const springBegin = new Date((await session.get(`${currentYear}01`)).SessionBeginDate);
+      const springEnd = new Date((await session.get(`${currentYear}01`)).SessionEndDate);
+      const summerBegin = new Date((await session.get(`${currentYear}05`)).SessionBeginDate);
+      const summerEnd = new Date((await session.get(`${currentYear}05`)).SessionEndDate);
+      const nextFallBegin = new Date((await session.get(`${currentYear}09`)).SessionBeginDate);
 
       setFirstDay(first);
       setLastDay(last);
       setCurrentSessionDescription(currentSessionDescription);
       setDaysRemaining(daysRemaining);
       setDaysFinished(daysInSemester - daysRemaining || 0);
       setLoading(false);
+      setFallEnd(fallEnd);
+      setSpringBegin(springBegin);
+      setSummerBegin(summerBegin);
+      setSummerEnd(summerEnd);
+      setSpringEnd(springEnd);
+      setFallBegin(fallBegin);
+      setNextFallBegin(nextFallBegin);
     };
 
     load();
   }, []);
 
-  const colors = theme360.colorSchemes.light.palette;
+  const academicTerms = ['Fall', 'Spring', 'Summer'];
+
+  const termBreaks = ['Winter', 'PreSummer', 'PostSummer'];
+
+  const msToDays = 1.15741e-8;
+  const winterBreak = Math.round((springBegin - fallEnd) * msToDays);
+  const daysRemainingWinter = Math.round((springBegin - today) * msToDays);
+  const winterProgress = (Math.round((today - fallEnd) * msToDays) / winterBreak) * 100;
+  const preSummer = Math.round((summerBegin - springEnd) * msToDays);
+  const preSummerProgress = (Math.round((today - springEnd) * msToDays) / preSummer) * 100;
+  const daysRemainingPreSummer = Math.round((summerBegin - today) * msToDays);
+  const postSummer = Math.round((nextFallBegin - summerEnd) * msToDays);
+  const daysRemainingPostSummer = Math.round((nextFallBegin - today) * msToDays);
+  const postSummerProgress = (Math.round((today - summerEnd) * msToDays) / postSummer) * 100;
```

Comment:
```suggestion
  const msToDays = 1.15741e-8;
  const winterBreak = Math.round((springBegin - fallEnd) * msToDays);
  const daysRemainingWinter = Math.round((springBegin - today) * msToDays);
  const winterProgress = (Math.round((today - fallEnd) * msToDays) / winterBreak) * 100;
  const preSummer = Math.round((summerBegin - springEnd) * msToDays);
  const preSummerProgress = (Math.round((today - springEnd) * msToDays) / preSummer) * 100;
  const daysRemainingPreSummer = Math.round((summerBegin - today) * msToDays);
  const postSummer = Math.round((nextFallBegin - summerEnd) * msToDays);
  const daysRemainingPostSummer = Math.round((nextFallBegin - today) * msToDays);
  const postSummerProgress = (Math.round((today - summerEnd) * msToDays) / postSummer) * 100;
```
Once again this is just a bit odd as well as the magic numbers.